code/modules/mob/living/carbon/alien/humanoid/alien_powers.dm

```diff
@@ -37,161 +41,370 @@ Doesn't work on other aliens/AI.*/
-				if (get_dist(src,M) <= 1)
-					M.adjustToxLoss(amount)
-					adjustToxLoss(-amount)
-					to_chat(M, "<span class='alien'>\The [src] has transfered [amount] plasma to you.</span>")
-					to_chat(src, "<span class='alien'>You have trasferred [amount] plasma to [M]</span>")
-				else
-					to_chat(src, "<span class='alien'>You need to be closer.</span>")
-	return
+	range = 7
+	spell_flags = WAIT_FOR_CLICK
+	var/storedmessage
+
+/spell/targeted/alienwhisper/channel_spell(mob/user = usr, skipcharge = 0, force_remove = 0)
+	if(!..()) //We only make it to this point if we succeeded in channeling or are removing channeling
+		return 0
+	if(user.spell_channeling && !force_remove)
+		storedmessage = sanitize(input("Message:", "Alien Whisper") as text|null)
+		if(!storedmessage) //They refused to supply a spell channeling
+			channel_spell(force_remove = 1)
+			return 0
+	else
+		storedmessage = null
+	return 1
+
+/spell/targeted/alienwhisper/is_valid_target(var/target)
+	if(!(spell_flags & INCLUDEUSER) && target == usr)
+		return 0
+	if(get_dist(usr, target) > range) //Shouldn't be necessary but a good check in case of overrides
+		return 0
+	return istype(target, /mob)
+
+/spell/targeted/alienwhisper/cast(var/list/targets, mob/user)
+	var/mob/M = targets[1]
+	if(!storedmessage) //Compatibility if someone reverts this to SELECTABLE from WAIT_FOR_CLICK
+		storedmessage = sanitize(input("Message:", "Alien Whisper") as text|null)
+		if(!storedmessage)
+			return 1
+	if(storedmessage)
+		var/turf/T = get_turf(user)
+		log_say("[key_name(user)] (@[T.x],[T.y],[T.z]) Alien Whisper: [storedmessage]")
+		to_chat(M, "<span class='alien'>You hear a strange, alien voice in your head... <em>[storedmessage]</span></em>")
+		to_chat(user, "<span class='alien'>You said: [storedmessage] to [M]</span>")
+
+/spell/targeted/alientransferplasma
+	name = "Transfer Plasma"
+	desc = "Transfer your plasma to another alien"
+	panel = "Alien"
+	hud_state = "alien_transfer"
+	override_base = "alien"
+
+	charge_type = Sp_HOLDVAR
+	holder_var_type = "plasma"
+	insufficient_holder_msg = "<span class='alien'>Not enough plasma stored.</span>"
+
+	range = 2
+	compatible_mobs = list(/mob/living/carbon/alien)
+
+//Does it take charge before casting? How to transfer to new alien
+/spell/targeted/alientransferplasma/cast(var/list/targets, mob/user)
+	var/mob/living/carbon/alien/M = targets[1]
+	var/amount = input(user, "Amount:", "Transfer Plasma to [M]") as num
+	if(amount)
+		amount = abs(round(amount))
+		holder_var_amount = amount
+		if(check_charge(user = user) && get_dist(user, M) <= range) //Since input is a blocking operation
+			take_charge(user = user)
+			to_chat(M, "<span class='alien'>\The [user] has transfered [amount] plasma to you.</span>")
+			to_chat(user, "<span class='alien'>You have trasferred [amount] plasma to [M]</span>")
+		else
+			to_chat(user, "<span class='alien'>You need to be closer.</span>")
+	holder_var_amount = 0
+
+/spell/targeted/alienneurotoxin
```

Comment:
This spell would be perfect as a `/spell/targeted/projectile` rather than having all the custom target code.